Update macOS install script and support multiple types of GPA (old)

kdk1616 opened the pull request:
Closes #83. Closes #71. Fixed mac install and start scripts (removed redis from start script)… Made function for toggling between weighted, unweighted, and % GPA (only works for current quarter right now... Also, There is a bug where if you select another quarter you cannot return to the initial quarter and have it work properly...).

kdk1616 committed:
Fixed mac install and start scripts (removed redis from start script). Made function for toggling between weighted, unweighted, and % GPA (only works for current quarter right now... Also, There is a bu where if you select another quarter you cannot return to the initial quarter and have it work properly...).

psvenk committed:
Revert changes to README

Apparently kdk1616's editor completely changed the formatting of
the README. As there were no substantial changes, I have reverted
all changes that were made to the README.

psvenk committed:
Merge branch 'master' into master

kdk1616 committed:
Fixed GPA Toggle not working for other quarters... had to hardcode variable for current term

psvenk commented:
Looks good, but I found a few small issues, such as:
- [ ] GPA indicator should be merged with GPA type selector, not with term dropdown
- [ ] A GPA of 4.0, 3.0, etc. should be displayed with at least one place after the decimal point
- [ ] The non-percent GPA formats do not play well with "Calculated GPA"

psvenk committed:
Use function for getting GPA out of 4.0

There is a function in extraFunctions.js to convert a percentage to
a GPA out of 4.0; the inline code duplicating that behavior has been
replaced by a function call.

psvenk committed:
Use tableData.terms.current for current term

This is a minor change, as the currently selected term is the same as
the current term when Aspine is first loading, but it aids code
readability for the refactoring to come.

psvenk committed:
Use for..of loop instead of iterator in GPA calculation

Abstracting away the iterator reduces the potential for errors and
simplifies the code.

psvenk committed:
Improve code readability

psvenk committed:
Further improve code readability and formatting

psvenk committed:
Normalize whitespace and comment out `console.log`s

psvenk committed:
Change GPA to an object + reduce dependence on currentQuarterRightNow

The GPA and calcGPA attributes of each term's object under tableData are
now themselves objects with attributes "percent", "outOfFour", and
"outOfFive". This approach is more scalable and also allows all GPAs
to be returned by a single function, computeGPA, which has been modified
to compute all three formats of GPA.

psvenk committed:
Refactor GPAType so that currentQuarterRightNow is not needed

The current quarter need not be hardcoded anymore.
Also, the refactored GPAType uses jQuery and is more readable.

psvenk committed:
Round 4.0 and 5.0 scale GPAs to 2 places after decimal point

psvenk committed:
Fix bug in which GPA switcher does not work

Because the selection of the <select> element is a string and it is
being compared to the number zero, I have changed triple equals
to double equals in order to invoke type coercion.

psvenk commented:
From what I've seen from editing the code the past two days, it seems that it may take around a week for this to be ready to merge.

kdk1616 committed:
Merge remote-tracking branch 'upstream/master'

kdk1616 committed:
Finished GPA types feature, added element to drop down with cumulative GPAs

kdk1616 committed:
Made login button autofocus, made cum GPA limited to two decimals
